### PR TITLE
data/systemd/snapd.run-from-snap: ensure snapd tooling is available

### PR DESCRIPTION
**Pull request by mvo5**
The tooling in /usr/lib/snapd must be made available to snapd.
This is archived by just bind mounting the snapd snaps tooling
dir (/usr/lib/snapd) into the system /usr/lib/snapd dir.

### CODE REVIEW

File: data/systemd/snapd.run-from-snap [hunk 1]
```diff
@@ -34,6 +34,8 @@ case $CMD in
             echo "cannot run without the prefix for the snapd snap"
             exit 1
         fi
+        echo "Add the bind mount of the snapd tooling"
+        mount -o bind "$PREFIX"/usr/lib/snapd /usr/lib/snapd
```
Comment (zyga) [inline data/systemd/snapd.run-from-snap]: Could this be a mount unit instead?

Comment (mvo5) [reply data/systemd/snapd.run-from-snap]: What is the advantage here? It seems like its more complex in terms of state to manage, i.e. the script needs to write one, clean out stale ones etc?

Comment (zyga) [reply data/systemd/snapd.run-from-snap]: It is more discoverable and explicit (less magic is good).

Question: does systemd shut down properly with this bind mount in place?

Comment (bboozzoo) [reply data/systemd/snapd.run-from-snap]: I think one could be generated and dumped into `/run/systemd/system`. Then the units that are generated just below could have proper `Requires=<that>.mount` snipped added.

File: data/systemd/snapd.run-from-snap [hunk 2]
```diff
@@ -34,13 +34,29 @@ case $CMD in
...
         fi
+        echo "Add the bind mount unit of the snapd tooling"
+        cat >/run/systemd/system/usr-lib-snapd.mount <<EOF
+[Unit]
+Description=Make the snapd snap tooling available for the system
+Before=snapd.service
+
+[Mount]
+What=$PREFIX/usr/lib/snapd
+Where=/usr/lib/snapd
```
Comment (zyga) [inline data/systemd/snapd.run-from-snap]: Do we need to take libexec into account?

Comment (mvo5) [reply data/systemd/snapd.run-from-snap]: Thanks for checking. This code only runs on ubuntu-core systems, so we can hardcode the (known) libexecdir here.

Comment (Conan-Kudo) [reply data/systemd/snapd.run-from-snap]: So then we'll have to fix this when there's more than Ubuntu Core systems? :/

**Pull request merged**